src/ape/api/providers.py

```diff
@@ -850,6 +850,32 @@ def send_call(self, txn: TransactionAPI, **kwargs) -> bytes:
             str: The result of the transaction call.
         """
 
+        if (
+            not isinstance(self, TestProviderAPI)
+            or not self.chain_manager._reports.track_gas
+            or not txn.receiver
+        ):
+            return self._send_call(txn, **kwargs)
+
+        # Handle call tree, if trace handlers set.
+        # For example, this is used for capturing gas usage in view calls
+        # during tests when the `--gas` flag was set.
+
+        try:
+            with self.chain_manager.isolate():
```

Comment:
- [ ] Figure out why this part is really really really slow and see if can do anything about it.